crates/anvil/src/config.rs

```diff
@@ -85,6 +86,13 @@ const BANNER: &str = r"
      \__,_| |_| |_|   \_/   |_| |_|
 ";
 
+/// Object-safe trait that enables injecting extra precompiles when using
+/// `anvil` as a library.
+pub trait PrecompileFactory {
+    /// Returns a set of precompiles to extend the EVM with.
+    fn precompiles(&self) -> Vec<(Address, Precompile)>;
+}
```

Comment:
I want to move this to a new file, evm.rs in this crate

eventually I think we'll move this into the evm crate

but if we do store `Vec<(Address, Precompile)>` in NodeConfig like we're doing now, then we don't really need that trait at all

Reply:
My idea was that this trait might prove useful for future work toward having something close to `ConfigureEvm`, but maybe this doesn't justify having it.

---

At first, I tried to use the trait as a field of `NodeConfig` like this:

```rust
    pub extra_precompiles: Option<Box<dyn PrecompileFactory>>,
```

But since `NodeConfig` derives `Clone`, I'd have to have:

```rust
pub trait PrecompileFactory: Send + Sync + Unpin + Debug + Clone {
```

Which makes it not object-safe. Made it an `Arc` as a workaround, wdyt?